packages/agora-rtc-react/src/hooks/tracks.ts

```diff
@@ -127,3 +133,229 @@ export function useVolumeLevel(audioTrack?: IRemoteAudioTrack | ILocalAudioTrack
+      await resolvedClient.massUnsubscribe(unsubscribeList);
+      if (!isUnmounted) {
+        setTracks(nextTracks.current.slice());
+      }
+    }
+
+    return joinDisposers([
+      () => {
+        isUnmounted = true;
+      },
+      listen(resolvedClient, "user-published", (pubUser, pubMediaType) => {
+        if (users.find(user => user.uid === pubUser.uid) && pubMediaType === "audio") {
+          subscribe(pubUser);
+        }
+      }),
+      listen(resolvedClient, "user-unpublished", (pubUser, pubMediaType) => {
+        if (users.find(user => user.uid === pubUser.uid) && pubMediaType === "audio") {
+          unsubscribe(pubUser);
+        }
+      }),
+    ]);
+  }, [isConnected, resolvedClient, tracks, users]);
+  return tracks;
+}
+
+/**
+ * Auto-subscribe and get remote user video track.
+ * Unsubscribe track on unmount.
+ */
+export function useRemoteVideoTracks(
+  users: IAgoraRTCRemoteUser[] | undefined,
+  client?: IAgoraRTCClient | null,
+): IRemoteVideoTrack[] {
+  const resolvedClient = useRTCClient(client);
+  const [tracks, setTracks] = useState<IRemoteVideoTrack[]>([]);
+  const isConnected = useIsConnected();
+  const nextTracks = useRef<IRemoteVideoTrack[]>([]);
+
+  useAsyncEffect(async () => {
+    if (!users || !isConnected) return;
+    let isUnmounted = false;
+    const subscribe = async (user: IAgoraRTCRemoteUser) => {
+      try {
+        if (!user.videoTrack && users.some(({ uid }) => user.uid === uid)) {
+          await resolvedClient.subscribe(user, "video").catch(error => {
+            void error;
+          });
+          if (
+            user.videoTrack &&
+            !nextTracks.current.some(track => track.getUserId() === user.uid)
+          ) {
+            nextTracks.current.push(user.videoTrack);
+          }
+
+          nextTracks.current = nextTracks.current.map(track => {
+            if (
+              user.videoTrack &&
+              track.getUserId() === user.uid &&
+              track.getTrackId() !== user.videoTrack.getTrackId()
+            ) {
+              return user.videoTrack;
+            } else {
+              return track;
+            }
+          });
+          if (!isUnmounted) {
+            setTracks(nextTracks.current);
+          }
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    const unsubscribe = async (user: IAgoraRTCRemoteUser): Promise<void> => {
+      try {
+        if (users.some(({ uid }) => user.uid === uid)) {
+          if (!isUnmounted) {
+            nextTracks.current = nextTracks.current.filter(track => track.getUserId() !== user.uid);
+            console.log(nextTracks.current);
+            setTracks(nextTracks.current);
+          }
+
+          await resolvedClient.unsubscribe(user, "video");
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    users.map(user => {
+      if (!user.videoTrack && user.hasVideo) {
+        subscribe(user);
+      }
```

Comment:
What should happen when `user.videoTrack` exists i.e. the subscription was done somewhere else? I think it should be directly put into the `nextTracks`.

Reply:
According to my test, `users: IAgoraRTCRemoteUser[] | undefined` will change when switching layouts. Due to asynchronous reasons, `user.videoTrack` of users in the new effect will exist.
I cannot guarantee that old user's unsubscribe can be executed after this function, otherwise it cannot be directly added to in nextTracks